### PR TITLE
Try to limit nvidia GPU queries to included GPUs

### PR DESCRIPTION
**Pull request by jvrobert**
Not sure this is the best solution, but it's a solution for now, pretty niche case probably anyway.

### CODE REVIEW

File: frigate/util.py [hunk 1]
```diff
@@ -926,6 +926,11 @@ def get_nvidia_gpu_stats() -> dict[str, str]:
         "--format=csv",
     ]
 
+    if ("CUDA_VISIBLE_DEVICES" in os.environ):
```
Comment (NickM-27) [inline frigate/util.py:929]: You're going to need to handle the case where the value for this env variable is `ALL` in which case we should not append an --id (same for the other env check as well)

**Review by NickM-27 — CHANGES_REQUESTED** [review]

File: frigate/util.py (hunk 1 above)
Comment (NickM-27) [reply frigate/util.py:929]: Probably just only append it if it's an int

**Comment by NickM-27** [comment]
you need to fix the python checks by running `python3 -m black frigate`

File: frigate/util.py [hunk 2]
```diff
@@ -926,6 +926,11 @@ def get_nvidia_gpu_stats() -> dict[str, str]:
         "--format=csv",
     ]
 
+    if "CUDA_VISIBLE_DEVICES" in os.environ and os.environ["CUDA_VISIBLE_DEVICES"].isdigit():
+        nvidia_smi_command.extend(["--id", os.environ['CUDA_VISIBLE_DEVICES']])
+    elif "NVIDIA_VISIBLE_DEVICES" in os.environ and os.environ["NVIDIA_VISIBLE_DEVICES"].isdigit():
+        nvidia_smi_command.extend(["--id", os.environ['NVIDIA_VISIBLE_DEVICES']])
+
```
Comment (NickM-27) [inline frigate/util.py:933]:
```suggestion
    if (
        "CUDA_VISIBLE_DEVICES" in os.environ 
        and os.environ["CUDA_VISIBLE_DEVICES"].isdigit()
    ):
        nvidia_smi_command.extend(["--id", os.environ['CUDA_VISIBLE_DEVICES']])
    elif (
        "NVIDIA_VISIBLE_DEVICES" in os.environ 
        and os.environ["NVIDIA_VISIBLE_DEVICES"].isdigit()
    ):
        nvidia_smi_command.extend(["--id", os.environ['NVIDIA_VISIBLE_DEVICES']])

```

**Review by NickM-27 — CHANGES_REQUESTED** [review]

File: frigate/util.py [hunk 3]
```diff
@@ -926,6 +926,17 @@ def get_nvidia_gpu_stats() -> dict[str, str]:
...
+    if (
+        "CUDA_VISIBLE_DEVICES" in os.environ 
+        and os.environ["CUDA_VISIBLE_DEVICES"].isdigit()
+    ):
+        nvidia_smi_command.extend(["--id", os.environ['CUDA_VISIBLE_DEVICES']])
+    elif (
+        "NVIDIA_VISIBLE_DEVICES" in os.environ 
+        and os.environ["NVIDIA_VISIBLE_DEVICES"].isdigit()
+    ):
+        nvidia_smi_command.extend(["--id", os.environ['NVIDIA_VISIBLE_DEVICES']])
```
Comment (NickM-27) [inline frigate/util.py:938]:
```suggestion
    if (
        "CUDA_VISIBLE_DEVICES" in os.environ
        and os.environ["CUDA_VISIBLE_DEVICES"].isdigit()
    ):
        nvidia_smi_command.extend(["--id", os.environ["CUDA_VISIBLE_DEVICES"]])
    elif (
        "NVIDIA_VISIBLE_DEVICES" in os.environ
        and os.environ["NVIDIA_VISIBLE_DEVICES"].isdigit()
    ):
        nvidia_smi_command.extend(["--id", os.environ["NVIDIA_VISIBLE_DEVICES"]])
```

**Review by NickM-27 — CHANGES_REQUESTED** [review]

**Review by NickM-27 — APPROVED** [review]

**Review by blakeblackshear — APPROVED** [review]

**Pull request merged**